pkg/fission-cli/cmd/function/update_container.go

```diff
@@ -52,11 +52,15 @@ func (opts *UpdateContainerSubCommand) do(input cli.Input) error {
 
 func (opts *UpdateContainerSubCommand) complete(input cli.Input) error {
 	fnName := input.String(flagkey.FnName)
+
 	fnNamespace := input.String(flagkey.NamespaceFunction)
+	if input.String(flagkey.Namespace) != "default" {
+		fnNamespace = input.String(flagkey.Namespace)
+	}
```

Comment:
Mostly I see a pattern across all CLI changes. 

We can probably think of a function here.

```
func getResourceNamespace(input cli.Input,  deprecatedFlag string) string {
         check if deprecated flag is set, access value 
         check if namespace flag is set, access value
        if both are unset return the current namespace.
}
```

Reply:
done